elastic/e2e-testing
# Use the correct beats repo

## What does this PR do?

This PR improves how Metricbeat compose files are synced:

- The test suite should be as independent as possible, and not depend on additional commands to run before. Therefore, the beats repo is fetched every time the test suite with the possibility to opt out.

- The repo is cloned / fetched for the current branch always. Before one would need to remember to delete and re-clone the repo when switching branches, or risk having the wrong compose files.

- It will only clone / fetch the latest commit, not the entire history. This speeds up the operation considerably. 

- The functionality, now that is used in a different place, is moved out of the `cmd` package.

## Checklist


## How to test this changes

Run some scenario with different versions, check that `.op/git/beats` checks out the right branch.


Related to https://github.com/elastic/e2e-testing/issues/1108
